src/vsr/replica.zig

```diff
@@ -2266,18 +2376,157 @@ pub fn ReplicaType(
             }
         }
 
+        fn on_request_sync_manifest(self: *Self, message: *Message) void {
+            assert(message.header.command == .request_sync_manifest);
+            if (self.ignore_sync_request_message(message)) return;
+
+            assert(message.header.op == self.superblock.staging.vsr_state.commit_min);
+            assert(message.header.parent == self.superblock.staging.checkpoint_id());
+
+            self.send_sync_trailer(message.header.command, .{
+                .offset = message.header.request,
+                .replica = message.header.replica,
+            });
+        }
+
+        fn on_request_sync_free_set(self: *Self, message: *Message) void {
+            assert(message.header.command == .request_sync_free_set);
+            if (self.ignore_sync_request_message(message)) return;
+
+            assert(message.header.op == self.superblock.staging.vsr_state.commit_min);
+            assert(message.header.parent == self.superblock.staging.checkpoint_id());
+
+            self.send_sync_trailer(message.header.command, .{
+                .offset = message.header.request,
+                .replica = message.header.replica,
+            });
+        }
+
+        fn on_request_sync_client_sessions(self: *Self, message: *Message) void {
+            assert(message.header.command == .request_sync_client_sessions);
+            if (self.ignore_sync_request_message(message)) return;
+
+            assert(message.header.op == self.superblock.staging.vsr_state.commit_min);
+            assert(message.header.parent == self.superblock.staging.checkpoint_id());
+
+            self.send_sync_trailer(message.header.command, .{
+                .offset = message.header.request,
+                .replica = message.header.replica,
+            });
+        }
+
+        fn on_sync_manifest(self: *Self, message: *Message) void {
+            assert(message.header.command == .sync_manifest);
+            assert(message.header.replica < self.replica_count);
+            if (self.ignore_sync_response_message(message)) return;
+
+            const stage = &self.sync_stage.request_trailers;
```

Comment:
It does, but it can easily be dropped through a refactor. And it's harder for the eye to see that it's there.

I started doing this in the SuperBlock. It's a bit of pain/friction but also helps to limit taking aliases in the first place (increasing the number of variables in scope, increasing probability of stack copy). If we do take the alias, then make the ptr type explicit.